packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/media/ImageStreamReaderTest.java

```diff
@@ -0,0 +1,164 @@
+    int requestedStreamFormat = ImageFormat.NV21;
+    int result = ImageStreamReader.computeStreamImageFormat(requestedStreamFormat);
+    assertEquals(result, ImageFormat.YUV_420_888);
+  }
+
+  /**
+   * If we are requesting NV21, then the planes should be processed and converted to NV21 before
+   * being sent to dart. We make sure yuv420ThreePlanesToNV21 is called when we are requesting
+   */
+  @Test
+  public void onImageAvailable_parsesPlanesForNv21() {
+    // Dart wants NV21 frames
+    int dartImageFormat = ImageFormat.NV21;
+
+    ImageReader mockImageReader = mock(ImageReader.class);
+    ImageStreamReaderUtils mockImageStreamReaderUtils = mock(ImageStreamReaderUtils.class);
+    ImageStreamReader imageStreamReader =
+        new ImageStreamReader(mockImageReader, dartImageFormat, mockImageStreamReaderUtils);
+
+    ByteBuffer mockBytes = ByteBuffer.allocate(0);
+    when(mockImageStreamReaderUtils.yuv420ThreePlanesToNV21(any(), anyInt(), anyInt()))
+        .thenReturn(mockBytes);
+
+    // The image format as streamed from the camera
+    int imageFormat = ImageFormat.YUV_420_888;
+
+    // Mock YUV image
+    Image mockImage = mock(Image.class);
+    when(mockImage.getWidth()).thenReturn(1280);
+    when(mockImage.getHeight()).thenReturn(720);
+    when(mockImage.getFormat()).thenReturn(imageFormat);
+
+    // Mock planes. YUV images have 3 planes (Y, U, V).
+    Image.Plane planeY = mock(Image.Plane.class);
+    Image.Plane planeU = mock(Image.Plane.class);
+    Image.Plane planeV = mock(Image.Plane.class);
+
+    // Y plane is width*height
+    // Row stride is generally == width but when there is padding it will
+    // be larger. The numbers in this example are from a Vivo V2135 on 'high'
+    // setting (1280x720).
+    when(planeY.getBuffer()).thenReturn(ByteBuffer.allocate(1105664));
+    when(planeY.getRowStride()).thenReturn(1536);
+    when(planeY.getPixelStride()).thenReturn(1);
+
+    // U and V planes are always the same sizes/values.
+    // https://developer.android.com/reference/android/graphics/ImageFormat#YUV_420_888
+    when(planeU.getBuffer()).thenReturn(ByteBuffer.allocate(552703));
+    when(planeV.getBuffer()).thenReturn(ByteBuffer.allocate(552703));
+    when(planeU.getRowStride()).thenReturn(1536);
+    when(planeV.getRowStride()).thenReturn(1536);
+    when(planeU.getPixelStride()).thenReturn(2);
+    when(planeV.getPixelStride()).thenReturn(2);
+
+    // Add planes to image
+    Image.Plane[] planes = {planeY, planeU, planeV};
+    when(mockImage.getPlanes()).thenReturn(planes);
+
+    CameraCaptureProperties mockCaptureProps = mock(CameraCaptureProperties.class);
+    EventChannel.EventSink mockEventSink = mock(EventChannel.EventSink.class);
+    imageStreamReader.onImageAvailable(mockImage, mockCaptureProps, mockEventSink);
+
+    // Make sure we processed the frame with parsePlanesForNv21
+    verify(mockImageStreamReaderUtils).yuv420ThreePlanesToNV21(any(), anyInt(), anyInt());
```

Comment:
Can you verify the parameters for this call too?

Reply:
Fixed in c170fe0f7e2a22d1cfa6fb0fda9c77b085ff7c26

Reply:
Fixed in c170fe0f7e2a22d1cfa6fb0fda9c77b085ff7c26